### ISSUE TITLE
Window title isn't updated with file * marker immediately

### ISSUE THREAD

**Issue by eishiya** [mapeditor/tiled#3203]
When a file is marked as dirty due to a change, it gets a * in front of its name. When it's saved or the changes are undone, the * is removed. This change is instant on the tab name, but it seems that the *window* title only gets updated when the document is switched to, closed, opened, or saved, rather than immediately.

The window title should be updated at the same time as the tab title.

I am using version 2021.12.03 (test build for selections not marking files as dirty), but I do not know whether this problem stems from that version or if I just noticed it now because I'm paying more attention to the dirty markers.

**Comment by bjorn** [comment]
> I am using version 2021.12.03 (test build for selections not marking files as dirty), but I do not know whether this problem stems from that version or if I just noticed it now because I'm paying more attention to the dirty markers.

Yeah, this appears to be a regression in that change. The window title is updated when the "clean" property of the `QUndoGroup` changes:
```c++
    connect(undoGroup, &QUndoGroup::cleanChanged, this, &MainWindow::updateWindowTitle);
```
However, with the selections no longer marking as dirty, the clean state of the undo group can no longer be used, since it does not take into account the selection changes. This causes its state to change at different times, and hence the updating of the window title will not trigger at the right times.

I guess the easiest fix is to just remove this connection, and connect to the `modifiedChanged` signal of the currently active document instead.
